carapace-server/src/main/java/org/carapaceproxy/server/ClientConnectionHandler.java

```diff
@@ -91,12 +93,22 @@ public ClientConnectionHandler(
         this.onClientDisconnected = onClientDisconnected;
         this.backendHealthManager = backendHealthManager;
         this.requestsLogger = requestsLogger;
+        this.listenerHost = listenerHost;
+        this.listenerPort = listenerPort;
     }
 
     public SocketAddress getClientAddress() {
         return clientAddress;
     }
 
+    public String getListenerHost() {
+        return listenerHost;
```

Comment:
maybe this is not needed.

the listener host/port is an internal information useful only inside the local network, and usually the listener will be on 0.0.0.0 so this information cannot be used for constructing an URI